cli/flox/doc/flox-pull.md

```diff
@@ -11,33 +11,64 @@ flox-pull - pull environment from FloxHub
 # SYNOPSIS
 
 ```
-flox [<general-options>] pull
+# Pull a new environment into a directory
+flox [<general-options>] pull <owner>/<name>
      [-d=<path>]
-     [-r=<owner>/<name> | <owner>/<name> | [-f]]
      [-f]
      [-c]
+     [-g=<generation>]
+
+# Update an existing environment in a directory
+flox [<general-options>] pull
+     [-d=<path>]
+     [-f]
+
+# Fetch updates for a remote environment
+flox [<general-options>] pull -r <owner>/<name>
+     [-f]
 ```
 
 # DESCRIPTION
 
 Pull an environment from FloxHub and create a local reference to it,
 or, if an environment has already been pulled, retrieve any updates.
 
-When pulling an environment for the first time, `-d` specifies the directory
-in which to create that environment.
-The remote environment is specified in the form `<owner>/<name>`.
-It may optionally be preceded by `-r`,
-but `-r` is not necessary and is accepted simply for consistency with other
-environment commands.
-
-When pulling an environment that has already been pulled, `-d` specifies which
-environment to sync.
-If `-d` is not specified and the current directory contains an environment, that
-environment is synced.
-`-f` may be specified in this case, forcibly updating the environment
-locally even if there are local changes not reflected in the remote environment.
-`<owner>/<name>` may be specified in this case and will replace the environment
-with the specified environment.
+## Pulling a new environment (`<owner>/<name> [--dir <dir>]`)
+
+Create an environment in the current directory or the directory specified by
+the `--dir` flag, that is linked to the centrally managed environemnt
+`<owner>/<name>` on FloxHub.
+You can make changes locally and push them back with
+[`flox-push(1)`](./flox-push.md).
+
+Alternatively, the `--copy` flag allows you to create an environment,
+but does not link it to its upstream on FloxHub.
+Optionally, the `--generation <generation>` can be used to select a specific
+generation to create a copy of.
+
+## Updating an existing environment in a directory (`[--dir]`)
+
+Without a `<owner>/<name>` argument, updates an environment that has already
+been pulled into the current directory, or the directory specified by the
+`--dir` flag .
+
+`-f` may be specified to forcibly update the environment locally even if
+there are local changes not reflected in the remote environment.
+
+## Updating a remote environment (`--remote <owner>/<name>`)
+
+When using the `--remote` flag, commands will operate on a
+**central persistent local copy** of the environment.
```

Comment:
suggestion nonblocking: I think just saying we store it in cache is a bit more concrete and might be easier to understand, maybe:
```
When using the `--remote` flag, commands will operate on a
copy of the environment stored in Flox's cache directory.
```